### PR TITLE
Misc chart polish

### PR DESCRIPTION
**Pull request by tlrobinson**
* [x] Hovering labels always show first label of the series #3599 #3803
* [x] hover styling
* [x] try to pick sane default columns for scatter plot
* [x] show metric/series name to the tooltip #3899
* [x] scatter plot series labels #3640
* [x] "Goal" line is cut off on charts #3525
* [x] Series > 5 aren't hovered correctly #4129

### CODE REVIEW

**Comment by tlrobinson** [comment]
@metabase/core-developers Review please?

**Comment by tlrobinson** [comment]
@metabase/core-developers Ping

**Review by salsakran — APPROVED** [review]
LGTM 
Didn't look at the css too closely however.

File: frontend/src/metabase/lib/visualization_settings.js [hunk 1]
```diff
@@ -59,12 +59,21 @@ function getDefaultColumns(series) {
...
-        bubble: null
-    };
+    let dimensions = cols.filter(isDimension);
+    let metrics = cols.filter(isMetric);
+    if (dimensions.length === 2 && metrics.length < 2) {
+        return {
+            dimensions: [dimensions[0].name],
+            metrics: [dimensions[1].name],
+            bubble: metrics.length === 1 ? metrics[0].name : null
+        }
```
Comment (tlrobinson) [inline frontend/src/metabase/lib/visualization_settings.js]: I haven't really thought that through completely, but in general for scatter/bubble I think numeric dimensions are most suitable for X and Y axis, text/category dimensions for breakout, and metric for bubble size.

**Pull request merged**